### ISSUE TITLE
Display own IPFS address on printable pages

### ISSUE THREAD

**Issue by kirushik** [paritytech/banana_split#33]
Printing the IPFS address of the current version of BananaSplit on the shards would make it much more future-proof to store secrets in BS.
Unfortunately, we need to generate those addresses while offline, so just using ipfs.js won't cut it.
What we can do instead is to read the actual blob with the currently opened file, hash it and generate an IPFS content-based ID from it.

This would require some reverse-engineering of IPFS chunking and CID generation (currently team is not eager to provide any specs for CID generation, see [this forum thread](https://discuss.ipfs.io/t/how-to-calculate-file-directory-hash/777/16) for example).

Note: it might be better to avoid chunking here, since it really complicated things. If making the resulting binary to take under 256Kb might not be possible right away, using `--chunker=size-1048576` might be a workaround. (The instructions would need to be printed on the shards as well then, concisely.)

**Comment by kirushik** [comment]
There's now https://github.com/skogard/nebulus, which is possibly able to calculate CID of our FileBuffer without us having to reimplement the whole logic by hand.

**Comment by prybalko** [comment]
I think [IPNS](https://docs.ipfs.io/concepts/ipns/#example-ipns-setup-with-cli) address can be used here. It'll redirect to the latest file version automatically.

 Here is uploaded version: https://ipfs.io/ipfs/QmZ6LQDu42DyCEwS1Mv95nGFWme8PSK6ci3ZJNFPdb48wd#/

**Comment by kirushik** [comment]
@prybalko no, sorry, using IPNS here wouldn't be a desired feature, for the following reasons:
1) IPFS addresses are immutable, IPNS are mutable. The IPNS record's contents might change at some moment in the future (for one reason or another), while having a direct link to the _exact same version_ of the BananaSplit you've used to combine the shards acts as an additional safeguard against accidental regressions introduced later on (say, if we update a crypto-library we use).
2) It's not quite clear to me if IPNS records (which are not ephemereal, and need to be re-broadcast from time to time, IIUC) would even last that long (BananaSplit has "20 years after Parity is gone" as a resilience target).
In addition to that, IPFS is stateless: if all copies hosted in IPFS are lost — later on anyone with the exact same file can add it back, and it will be locate-able at the exact same link. Not sure what happens with IPNS record after the content it points to is gone and then re-added; hope it works, but don't want to risk this.
3) It introduces another point of Parity's control of the project: ownership of the publishing key used for IPNS. Project's goal is to have less of those, not more.
4) In practice, IPNS wasn't performing reliably for me the last time I tried (a year/year and a half ago I think?) Fingers crossed things have been improved since (IPFS seen several version bumps in the meantime), but don't have capacity to thoroughly test this.
